File: flutter_twilio_commkit_ios/ios/Classes/Core/TwilioSdkCredentials.swift
import Foundation

/// Holds the Twilio project credentials for the current SDK session.
/// Set once during initialize() and accessible by all managers.
final class TwilioSdkCredentials {

    /// Twilio Account SID — starts with `AC`.
    let accountSid: String

    /// API Key SID — starts with `SK`.
    let apiKeySid: String

    /// Push Credential SID — starts with `CR` (optional).
    let pushCredentialSid: String?

    /// TwiML Application SID — starts with `AP` (optional).
    let outgoingApplicationSid: String?

    init(
        accountSid: String,
        apiKeySid: String,
        pushCredentialSid: String? = nil,
        outgoingApplicationSid: String? = nil
    ) {
        self.accountSid = accountSid
        self.apiKeySid = apiKeySid
        self.pushCredentialSid = pushCredentialSid
        self.outgoingApplicationSid = outgoingApplicationSid
    }

    /// The current session credentials. `nil` until `initialize()` is called.
    static var current: TwilioSdkCredentials?
}

File: flutter_twilio_commkit_android/android/src/main/kotlin/com/twiliocommkit/android/core/TwilioSdkCredentials.kt
package com.twiliocommkit.android.core

/**
 * Holds the Twilio project credentials for the current SDK session.
 *
 * Set once during [FlutterTwilioCommKitAndroidPlugin.initialize] and
 * accessible by all managers via the [current] singleton reference.
 */
data class TwilioSdkCredentials(
    /** Twilio Account SID — starts with AC */
    val accountSid: String,
    /** API Key SID — starts with SK */
    val apiKeySid: String,
    /** Push Credential SID — starts with CR (optional) */
    val pushCredentialSid: String?,
    /** TwiML Application SID — starts with AP (optional, used for outgoing calls) */
    val outgoingApplicationSid: String?
) {
    companion object {
        /** Current session credentials. Null until initialize() is called. */
        @Volatile
        var current: TwilioSdkCredentials? = null
    }
}

